gethoney/main.py

```diff
@@ -1,49 +1,51 @@
-from fastapi import FastAPI
-from gethoney.models import Honeypot
-from gethoney.crud import conn, curr
-import requests
-
-app = FastAPI()
-
-curr.execute(
-    """CREATE TABLE IF NOT EXISTS honeypots
-    (id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, url TEXT, description TEXT)"""
-)
-
-# curr.execute(
-#     """CREATE TABLE IF NOT EXISTS logs
-#     (id INTEGER PRIMARY KEY, log_id INTEGER, name TEXT))"""
-# )
+from fastapi import Depends, FastAPI, HTTPException, Response
 
+from gethoney.crud import Database
+from gethoney.models import Honeypot
 
-@app.get("/")
-def read_root():
-    return {"Hello": "World"}
+db_path = "../data/gethoney.db"
 
+app = FastAPI()
 
-@app.post("/honeypots/", status_code=201)
-def create_honeypots(honeypots: list[Honeypot]):
 
-    curr.executemany(
-        """INSERT INTO honeypots
-                (name, url, description)
-                VALUES (?, ?, ?) """,
-        [(honeypot.name, honeypot.url, honeypot.description) for honeypot in honeypots],
-    )
-    conn.commit()
-    return honeypots
+def get_db():
+    yield Database(db_path)
 
 
-@app.get("/honeypots/")
-def get_honeypots():
-    curr.execute("SELECT * FROM honeypots")
-    data = curr.fetchall()
-    return data
+@app.get("/honeypots/{id}", status_code=200)
+def get_honeypot(id: int, db: Database = Depends(get_db)) -> Honeypot:
+    honeypot = db.read_honeypot(id)
+    if not honeypot:
+        raise HTTPException(status_code=404, detail="Honeypot not found.")
+    else:
+        return honeypot
 
 
-# @app.get("/honeypots/logs/")
-# def get_logs(honeypot: list[], )
+@app.get("/honeypots/", status_code=200)
+def list_honeypots(db: Database = Depends(get_db)) -> list[Honeypot]:
+    return db.list_honeypots()
 
 
-# @app.post("/honeypots/logs/")
-# def create_logs(honeypot: Honeypot):
+@app.post("/honeypots/", status_code=201)
+def create_honeypot(honeypot: Honeypot, db: Database = Depends(get_db)) -> Honeypot:
+    db.create_honeypot(honeypot)
+    return honeypot
+
+
+@app.delete("/honeypots/{honeypot_id}", status_code=200)
+def delete_honeypot(honeypot_id: int, db: Database = Depends(get_db)) -> Honeypot:
+    honeypot = db.read_honeypot(honeypot_id)
+    if not honeypot:
+        return Response(status_code=404)
+    else:
+        db.delete_honeypot(honeypot_id)
+        return Response(status_code=204)
+
+
+@app.put("/honeypots/{honeypot_id}", status_code=200)
+def update_honeypot(honeypot_id: int, update_data: Honeypot, db: Database = Depends(get_db)) -> Honeypot:
+    data = db.update_honeypot(update_data, honeypot_id)
+    if not data:
+        raise HTTPException(status_code=404, detail="Honeypot not found.")
+    else:
+        return data

```

Comment:
You should be responding with your Honeypot response model since you should be fetching the ID as well. for example, how would you know what honeyput to update if you dont have the corresponding honeypot ID?

Reply:
Ok, I see that your CRUD returns a honeypot response. You need to update the return types of these functions